### PR TITLE
A few updates to the getting started docs

### PR DESCRIPTION
**Pull request by natalieparellano**
Based on getting `e2e` working for the first time

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
+* [`pack`](https://buildpacks.io/docs/tools/pack/)
+* [`docker`](https://docs.docker.com/get-docker/)
+* [`ytt`](https://carvel.dev/ytt/docs/v0.44.0/install/)
+
+#### When using public registries
+
 ```bash
 kubectl cluster-info # ensure you have access to a cluster
-./hack/apply.sh <IMAGE/NAME> # <IMAGE/NAME> is a writable and publicly accessible location 
+docker login <registry namespace> # must be writable and publicly accessible; e.g., your Docker Hub username or gcr.io/<some-project>
```
Comment (natalieparellano) [inline DEVELOPMENT.md:19]: An image "name" such as some-docker-username/random-image causes the tests to fail

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -65,64 +76,29 @@ EOF
...
+IMAGE_REGISTRY=gcr.io/<some-project> \
+  IMAGE_REGISTRY_USERNAME=_json_key \
+  IMAGE_REGISTRY_PASSWORD=$(cat gcp.json) \
+  make e2e
 ```
 
-### Libgit2 v1.3.0 Dependency and Installation
+* The IMAGE_REGISTRY environment variable must point at a registry with local write access - e.g.
 
-Several unit tests depend upon libgit2 v1.3.0.
```
Comment (natalieparellano) [inline DEVELOPMENT.md:92]: This seems to no longer be true

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -65,64 +76,29 @@ EOF
...
-```bash
-export KPACK_TEST_NAMESPACE_LABELS="istio-injection=disabled,purpose=test"
-```
-
-* The IMAGE_REGISTRY environment variable must point at a registry with local write access 
+* 🍿 These tests can take anywhere from 20-30 minutes depending on your setup
 
 ```bash
-IMAGE_REGISTRY=gcr.io/<some-project> go test ./test/...
+IMAGE_REGISTRY=gcr.io/<some-project> \
```
Comment (natalieparellano) [inline DEVELOPMENT.md:88]: Side note, while `./hack/apply.sh` works with index.docker.io I couldn't get the end-to-end tests to pass this way (they failed with "unauthorized" error). I recall there are some oddities WRT ggcr/docker and Docker Hub specifically, but I didn't dig too much here.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -65,64 +76,29 @@ EOF
 ### Running Unit Tests
 
 ```bash
-go test ./pkg/...
+make unit
```
Comment (natalieparellano) [inline DEVELOPMENT.md:79]: Using `make` allows us to inherit the timeout which I hit on one of the suites

File: DEVELOPMENT.md (hunk 3 above)
Comment (tomkennedy513) [reply DEVELOPMENT.md:88]: ya there are some weirdisms but I forget the exact config needed for running the e2e tests against dockerhub

**Review by tomkennedy513 — APPROVED** [review]

**Pull request merged**